### PR TITLE
npm script improvements

### PR DESCRIPTION
**Pull request by fleupold**
1) Proper error propagating through use of try catch block and calling callback with error
2) Adding Usage information 
3) Remove eslint-disable
4) using `waitForNBlock` in mine_blocks script (addresses #34)

### CODE REVIEW

**Review by josojo — APPROVED** [review]
Looks good! Thanks

File: scripts/deposit.js [hunk 1]
```diff
@@ -1,38 +1,41 @@
-/* eslint-disable no-unused-vars */
-/* eslint-disable indent */
 const SnappBase = artifacts.require("SnappBase")
 const ERC20Mintable = artifacts.require("ERC20Mintable.sol")
 
 const zero_address = "0x0000000000000000000000000000000000000000"
```
Comment (bh2smith) [inline scripts/deposit.js]: I know I put this here, but recently realized this can be replaces with `0x0` (without quotes)

Comment (bh2smith) [inline scripts/deposit.js]:
```suggestion
const zero_address = 0x0
```

**Review by bh2smith — APPROVED** [review]
This looks fantastic! Thanks for learning me some JS!

File: scripts/apply_deposits.js [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
-        console.log("Warning: Requested deposit slot has already been applied")
-        callback()
+      callback("Error: Requested deposit slot has already been applied")
     }
 
     console.log("Current slot for :", slot , " with curr_state", curr_state, " and new_state", new_state)
-    const tx = await instance.applyDeposits(slot, curr_state, new_state, deposit_state.shaHash)
+    await instance.applyDeposits(slot, curr_state, new_state, deposit_state.shaHash)
     const deposit_state2 = await instance.deposits.call(slot)
     console.log("New appliedAccountStateIndex is:",deposit_state2.appliedAccountStateIndex )
```
Comment (bh2smith) [inline scripts/apply_deposits.js:23]: Do we not need es-lint disable for the console.log? Did we disable this outside of the file?

File: scripts/setup_environment.js [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-
+const { setupEnvironment } = require("../test/utilities.js")
 
 module.exports = async function(callback) {
+  try {
     const instance = await SnappBase.deployed()
     const accounts = await web3.eth.getAccounts()
     const token_owner = accounts[1]
-
+
```
Comment (bh2smith) [inline scripts/setup_environment.js]:
```suggestion

```

File: scripts/apply_deposits.js (hunk 2 above)
Comment (fleupold) [reply scripts/apply_deposits.js:23]: The es-lint disable that I removed only affected indentation and unused variables, no console.log.

Comment (bh2smith) [reply scripts/apply_deposits.js:23]: Weird. I am surprised to see that my editor is no longer complaining about console.log

**Pull request merged**